packages/odyssey-react-mui/src/labs/SideNav.tsx

```diff
@@ -722,7 +761,16 @@ const SideNav = ({
                   </SideNavListItemContainer>
                 );
               } else {
-                return <SideNavItemContent key={item.id} {...item} />;
+                if (item.isSelected && !sideNavItemIdToScrollTo) {
+                  setSideNavItemIdToScrollTo(item.id);
+                }
+                return (
+                  <SideNavItemContent
+                    key={item.id}
+                    ref={scrollToNodeRefIfApplicable(item.id)}
```

Comment:
Personally, I'd prefer to be slightly more verbose here. Just so it's super clear what is happening. 
```js
ref={(element) => scrollToNodeRefIfApplicable(childProps.id, element)}
```

Reply:
Are we passing a function to `ref`? I'm not sure I understand what's happening. `ref` takes an object. Passing functions to `ref` will be removed in React 19 and should be already deprecated.

Just pass the ref and use `useEffect` to do the scroll.